app/Auth/AlbumAuthorisationProvider.php

```diff
@@ -490,24 +491,21 @@ private function getUnlockedAlbumIDs(): array
 	 */
 	public function isEditable(?AbstractAlbum $album): bool
 	{
-		if (AccessControl::is_admin()) {
-			return true;
-		}
-		if (!AccessControl::is_logged_in()) {
+		if (!Auth::check()) {
 			return false;
 		}
-
-		$user = AccessControl::user();
-
-		if (!$user->may_upload) {
+		if (Gate::check('admin')) {
+			return true;
+		}
+		if (!Gate::check('can-upload')) {
 			return false;
 		}
 
 		// The root album and smart albums get a pass
 		return
 			$album === null ||
 			$album instanceof BaseSmartAlbum ||
-			($album instanceof BaseAlbum && $album->owner_id === $user->id);
+			($album instanceof BaseAlbum && $album->owner_id === Auth::authenticate()->id);
```

Comment:
I am in favor for the gates, especially for the `before()` mechanism which allows to focus on the actual cases of the access rather than the exceptions such as admin.

I also rewrote the admin definition in UserPolicy such that if we want to extend this, we can do it more easilly in a single place.